fal/controllers/anime_stats.py

```diff
@@ -20,221 +31,237 @@
-
-
-def get_anime_simulcast_region_counts(
-    simulcast_lines: Optional[Iterable[str]],
-    session: Session
-) -> Dict[int, int]:
-    print("Getting region counts of each anime in simulcast file")
-    anime_simulcast_region_counts = {}
-    if simulcast_lines is not None:
-        for line in simulcast_lines:
-            title, subs = line.split("=")
-            title = title.strip()
-            anime = Anime.get_anime_from_database_by_name(title, session)
-            if anime is None:
-                print(f"{title} is not found in database")
-            else:
                 num_regions = len(
                     [entry for entry in subs.split() if entry == "simul"]
                 )
-                anime_simulcast_region_counts[anime.id] = num_regions
-    return anime_simulcast_region_counts
-
-
-def get_licensed_anime(licenses_lines: Optional[Iterable[str]], session: Session) -> Set[int]:
-    print("Getting licensed anime from licenses file")
-    licensed_anime = set()
-    if licenses_lines is not None:
-        for title in licenses_lines:
-            anime = Anime.get_anime_from_database_by_name(title.strip(), session)
-            if anime is None:
-                print(f"{title} is not found in database")
-            else:
-                licensed_anime.add(anime.id)
-    return licensed_anime
-
-
-def populate_anime_weekly_stats(
-    simulcast_lines: Optional[Iterable[str]] = None,
-    licenses_lines: Optional[Iterable[str]] = None,
-) -> None:
-    """
-    Populates the AnimeWeeklyStat table with a row for each anime
-    using data from Jikan.
-    """
-
-    season_of_year = config.get("season info", "season").lower()
-    year = config.getint("season info", "year")
-    week = config.getint("weekly info", "current-week")
-
-    if is_week_to_calculate("scoring.simulcast", week) and simulcast_lines is None:
-        raise ValueError(f"simulcast file is required for week {week}")
-    if is_week_to_calculate("scoring.license", week) and licenses_lines is None:
-        raise ValueError(f"licenses file is required for week {week}")
-
-    with session_scope() as session:
-        anime_simulcast_region_counts = get_anime_simulcast_region_counts(simulcast_lines, session)
-        licensed_anime = get_licensed_anime(licenses_lines, session)
-
-
-        season = Season.get_season_from_database(season_of_year, year, session)
-        anime_list = cast(Iterable[Anime], season.anime)
+                anime_simulcast_region_counts[anime.mal_id] = num_regions
+        return anime_simulcast_region_counts
+
+    def get_licensed_anime(self, session: Session) -> Set[int]:
+        print("Getting licensed anime from licenses file")
+        licensed_anime = set()
+        if self.licenses_lines is not None:
+            for title in self.licenses_lines:
+                title = title.strip()
+                anime = Anime.get_by_name(title, session)
```

Comment:
I don't think this line should be here since it's inside the try/catch statement.

Reply:
derp